azurelinuxagent/ga/exthandlers.py

```diff
@@ -265,6 +279,13 @@ def __init__(self, protocol):
         self.last_etag = None
         self.log_report = False
         self.log_process = False
+        # The GoalState Aggregate status needs to report the last status of the GoalState. Since we only process
+        # extensions on incarnation change, we need to maintain its state.
+        # Setting the status as Initialize here. This would be overridden as soon as the first GoalState is processed
+        # (once self._extension_processing_allowed() is True).
+        self.__gs_aggregate_status = GoalStateAggregateStatus(status=GoalStateState.Initialize, seq_no="-1",
```

Comment:
It would make more sense to save the whole json status to disk, the task that @kevinclark19a was doing at some point but I think he got side tracked  with something else.

Reply:
can we do it as part of this feature? (can be a different PR) This will definitively be useful for  debugging

Reply:
Yeah makes sense. @kevinclark19a if you're busy with other stuff I can take this over (let's sync about this offline too)